src/envoy/http/service_control/client_cache.cc

```diff
@@ -304,12 +307,25 @@ CancelFunc ClientCache::callCheck(
       request, response,
       [this, response, on_done](const Status& status) {
         CheckResponseInfo response_info;
```

Comment:
FYI there is a bug here that I will create a PR for. I'll also re-order the logic here, it should be easier for you to track stats then.